### PR TITLE
[ADD] xodoo: init

### PR DESCRIPTION
**Pull request by oerp-odoo**
``xodoo`` is collection of various scripts intended to be used for odoo. It is based on ``click-odoo`` to connect and manage odoo.

Scripts
~~~~~~~

xodoo-migrate
-------------

Migration tool intended to run various migration scripts for Odoo. It should be used when migration scripts directly on modules do not make sense (e.g. need to generate some customer specific data that is not directly related with any module).

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
```
Comment (SButko) [inline tests/conftest.py]: Probably should update to yours or girteka?

File: tests/test_migration.py [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+def test_03_migrate_fail(odoodb, path_test_data):
+    # GIVEN
+    path_mig_2 = path_test_data / "mig2"
+    # WHEN, THEN
+    with OdooEnvironment(database=odoodb, rollback=True) as env:
+        cr = env.cr
+        with pytest.raises(
+            click.ClickException, match=r"Something went wrong migrating '.+':"
+        ):
+            migration.migrate(env, str(path_mig_2))
```
Comment (SButko) [inline tests/test_migration.py:93]: Why it should fail on ```env.ref('base.res_partner_category_11').name = 'mig-services-111'```?

**Review by SButko — APPROVED** [review]

File: tests/test_migration.py (hunk 2 above)
Comment (oerp-odoo) [reply tests/test_migration.py:93]: It won't fail. Maybe test is confusing. I only test if it was not changed. When you get an exception on any of migration scripts, everything is rollbacked.

Comment (oerp-odoo) [reply tests/test_migration.py:93]: Also note, that test is running both scripts and one of them is intended to fail. So whole transaction is rollbacked.

**Pull request merged**